app/javascript/chat/ReportAbuse/index.jsx

```diff
@@ -0,0 +1,148 @@
+import { h, Fragment } from 'preact';
+import PropTypes from 'prop-types';
+import { useState } from 'preact/hooks';
+import { reportAbuse, blockUser } from '../actions/requestActions';
+import { addSnackbarItem } from '../../Snackbar';
+import { Button, FormField, RadioButton } from '@crayons';
+
+/**
+ * This component render the report abuse
+ *
+ * @param {object} props
+ * @param {object} props.data
+ * @param {function} props.closeReportAbuseForm
+ *
+ * @component
+ *
+ * @example
+ *
+ * <ReportAbuse
+ *  data={data}
+ *  closeReportAbuseForm={closeReportAbuseForm}
+ * />
+ *
+ */
+function ReportAbuse({ data, closeReportAbuseForm }) {
+  const [category, setCategory] = useState(null);
+
+  const handleChange = (e) => {
+    setCategory(e.target.value);
+  };
+
+  const handleSubmit = async () => {
+    const response = await reportAbuse(
+      data.message,
+      'connect',
+      category,
+      data.user_id,
+    );
+    const { success, message } = response;
+    if (success) {
+      const confirmBlock = window.confirm(
+        `The message will be reported.\n\nWould you like to block this person as well?\n\nThis will:
+      - prevent them from commenting on your posts
+      - block all notifications from them
+      - prevent them from messaging you via chat`,
+      );
```

Comment:
during testing I found this part of the flow not very intuitive - because the confirm popup only surfaces OK/cancel buttons it feels a bit like we're asking the user whether they want to go ahead with the report or cancel it (rather than 'OK yes please block' vs 'no thanks'.

Maybe not in design/scope for this ticket, I think it would be cool if this question was part of the main report form so we don't need the popup at all (future optimisation?)

Reply:
I think that's a great suggestion that we could improve in another PR. I have no specs for this PR, so I think it was what they thought was good.

Reply:
Mind creating an issue for this @aitchiss and link it to this PR? Thanks!